services/frontend/src/components/visibilityConfirmation/VisibilityConfirmationView.jsx

```diff
@@ -0,0 +1,112 @@
+import PropTypes from "prop-types";
+import { Descriptions, Modal, Tag } from "antd";
+import { FormattedMessage } from "react-intl";
+
+const VisibilityConfirmationView = ({
+  visibleCards,
+  visible,
+  onOk,
+  onCloseModal,
+}) => {
+  const getCardStatusElement = (cardStatus) => {
+    switch (cardStatus) {
+      case "PRIVATE":
+        return (
+          <>
+            <Tag color="red">
+              <FormattedMessage id="visibility.card.private" />
+            </Tag>
+          </>
+        );
+      case "PUBLIC":
```

Comment:
can this be changed to a function with variables being passed in for each case since its repeated code